### PR TITLE
nixosTests.freenet: init

### PR DESCRIPTION
**Pull request by nagy**
###### Description of changes

This is supposed to test basic functionality of freenet. I want to enhance this so that two nodes connect to each other and exchange files, but for this, I need to finish the building from source of freenet, which I dont want to work on without a basic test. So lets start with this simple test for now.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/freenet.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  };
+
+  nodes = {
+    machine = { ... }: {
+      services.freenet.enable = true;
+    };
+  };
+
+  testScript = ''
+    machine.start()
```
Comment (NickCao) [inline nixos/tests/freenet.nix:14]: machine is automatically started on first use, this is not needed.

File: nixos/tests/freenet.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    machine.wait_for_unit("freenet.service")
+    machine.wait_for_open_port(8888)
+    machine.wait_until_succeeds("curl -sfL http://localhost:8888/ | grep Freenet")
+    machine.succeed("systemctl stop freenet")
```
Comment (NickCao) [inline nixos/tests/freenet.nix:18]: leave it running won't hurt.

File: nixos/tests/freenet.nix (hunk 1 above)
Comment (nagy) [reply nixos/tests/freenet.nix:14]: I have mostly seen this in other tests and just copied, but I will take it out.

File: nixos/tests/freenet.nix (hunk 2 above)
Comment (NickCao) [reply nixos/tests/freenet.nix:18]: That's a reasonable concern.

**Comment by nagy** [comment]
updated.

**Review by NickCao — APPROVED** [review]

**Comment by marsam** [comment]
@GrahamcOfBorg test freenet

**Pull request merged**